src/overlays/actors/ovl_Elf_Msg2/z_elf_msg2.c

```diff
@@ -31,32 +30,120 @@ const ActorInit Elf_Msg2_InitVars = {
     (ActorFunc)NULL,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_8096F0B0[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_VEC3F_DIV1000(scale, 200, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneForward, 1000, ICHAIN_STOP),
 };
 
-#endif
-
-extern InitChainEntry D_8096F0B0[];
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Elf_Msg2/D_8096F0C0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Elf_Msg2/ElfMsg2_SetupAction.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Elf_Msg2/func_8096EC4C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Elf_Msg2/ElfMsg2_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Elf_Msg2/ElfMsg2_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Elf_Msg2/func_8096EE50.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Elf_Msg2/func_8096EE64.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Elf_Msg2/func_8096EF98.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Elf_Msg2/func_8096EFD0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Elf_Msg2/ElfMsg2_Update.s")
+void ElfMsg2_SetupAction(ElfMsg2* this, ElfMsg2ActionFunc actionFunc) {
+    (void)"共倒れ";   // Collapse together
+    (void)"共倒れ２"; // Collapse 2
+    (void)"共倒れ";
+    this->actionFunc = actionFunc;
+}
+
+s32 func_8096EC4C(ElfMsg2* this, PlayState* play) {
+    if ((this->actor.home.rot.y > 0) && (this->actor.home.rot.y < 0x81) &&
+        (Flags_GetSwitch(play, this->actor.home.rot.y - 1))) {
+        if (ELFMSG2_GET_8(&this->actor) != 0x7F) {
+            Flags_SetSwitch(play, ELFMSG2_GET_8(&this->actor));
+        }
+        Actor_MarkForDeath(&this->actor);
+        return true;
+    }
+    if (this->actor.home.rot.y == 0x81) {
+
+        if (Flags_GetClear(play, this->actor.room)) {
+            if (ELFMSG2_GET_8(&this->actor) != 0x7F) {
```

Comment:
```suggestion
        if (Flags_GetClear(play, this->actor.room)) {
            (void)"共倒れ２";
            if (ELFMSG2_GET_8(&this->actor) != 0x7F) {
```
Probably put the 2 here based on OoT